openslides/assignments/views.py

```diff
@@ -242,6 +243,134 @@ def get_serializer_class(self):
             serializer_class = AssignmentShortSerializer
         return serializer_class
 
+    @detail_route(methods=['post', 'delete'])
+    def candidature_self(self, request, pk=None):
+        """
+        View to nominate self as candidate (POST) or withdraw own candidature
+        (DELETE).
+        """
+        if not request.user.has_perm('assignments.can_nominate_self'):
+            self.permission_denied(request)
+        assignment = self.get_object()
+        if assignment.is_elected(request.user):
+            raise ValidationError({'detail': _('You are already elected.')})
+        if request.method == 'POST':
+            message = self.nominate_self(request, assignment)
+        else:
+            # request.method == 'DELETE'
+            message = self.withdraw_self(request, assignment)
+        return Response({'detail': message})
+
+    def nominate_self(self, request, assignment):
+        if assignment.phase == assignment.PHASE_FINISHED:
+            raise ValidationError({'detail': _('You can not candidate to this election because it is finished.')})
+        if assignment.phase == assignment.PHASE_VOTING and not request.user.has_perm('assignments.can_manage'):
+            # To nominate self during voting you have to be a manager.
+            self.permission_denied(request)
+        if assignment.is_candidate(request.user):
+            raise ValidationError({'detail': _('You are already a candidate of this election.')})
```

Comment:
I would prefer to remove the two lines. If you update a nodes content with new content, and the new content is exactly the same as the old one, you would also expect to get a 200. Or if you safe a file that has not changed it would be strange to get an IO error.

So when you update your candidate-status to the status you already have, you should not get an error